# These settings must match:
# - Port numbers specified in the ABB robot controller's system configurations for EGM communication.
# - Names of the mechanical unit groups intended to be controlled by respective EGM communication channel.
egm:
  channel_1:
    port_number: 6511
    mech_unit_group: "rob_l"
  channel_2:
    port_number: 6512
    mech_unit_group: "rob_r"

# These settings must match:
# - Joint names extracted from the ABB robot controller.
# - The 'yumi_' prefix must match the nickname/identifier,
#   which is set in the corresponding launch file.
#
# See http://wiki.ros.org/joint_limits_interface for more info about available joint limits:
joint_limits:
  yumi_robl_joint_1:
    min_position: -2.84
    max_position: 2.84
    has_velocity_limits: true
    max_velocity: 3.14159265359
  yumi_robr_joint_1:
    min_position: -2.84
    max_position: 2.84
    has_velocity_limits: true
    max_velocity: 3.14159265359
  yumi_robl_joint_2:
    min_position: -2.4
    max_position: 0.65
    has_velocity_limits: true
    max_velocity: 3.14159265359
  yumi_robr_joint_2:
    min_position: -2.4
    max_position: 0.65
    has_velocity_limits: true
    max_velocity: 3.14159265359
  yumi_robl_joint_4:
    min_position: -2
    max_position: 1.29
    has_velocity_limits: true
    max_velocity: 3.14159265359
  yumi_robr_joint_4:
    min_position: -2
    max_position: 1.29
    has_velocity_limits: true
    max_velocity: 3.14159265359
  yumi_robl_joint_5:
    min_position: -4.9
    max_position: 4.9
    has_velocity_limits: true
    max_velocity: 6.98131700798
  yumi_robr_joint_5:
    min_position: -4.9
    max_position: 4.9
    has_velocity_limits: true
    max_velocity: 6.98131700798
  yumi_robl_joint_6:
    min_position: -1.43
    max_position: 2.3
    has_velocity_limits: true
    max_velocity: 6.98131700798
  yumi_robr_joint_6:
    min_position: -1.43
    max_position: 2.3
    has_velocity_limits: true
    max_velocity: 6.98131700798
  yumi_robl_joint_7:
    min_position: -3.89
    max_position: 3.89
    has_velocity_limits: true
    max_velocity: 6.98131700798
  yumi_robr_joint_7:
    min_position: -3.89
    max_position: 3.89
    has_velocity_limits: true
    max_velocity: 6.98131700798
  yumi_robl_joint_3:
    min_position: -2.84
    max_position: 2.84
    has_velocity_limits: true
    max_velocity: 3.14159265359
  yumi_robr_joint_3:
    min_position: -2.84
    max_position: 2.84
    has_velocity_limits: true
    max_velocity: 3.14159265359
